core/state/statedb.go

```diff
@@ -84,96 +47,9 @@ func (wp *workerPool) Done() {
 	wp.BoundedWorkers.Wait()
 }
 
-func WithConcurrentWorkers(prefetchers int) ethstate.PrefetcherOption {
+func WithConcurrentWorkers(prefetchers int) state.PrefetcherOption {
 	pool := &workerPool{
 		BoundedWorkers: utils.NewBoundedWorkers(prefetchers),
 	}
-	return ethstate.WithWorkerPools(func() ethstate.WorkerPool { return pool })
-}
-
-// Retrieve the balance from the given address or 0 if object not found
-func (s *StateDB) GetBalanceMultiCoin(addr common.Address, coinID common.Hash) *big.Int {
-	NormalizeCoinID(&coinID)
-	return s.StateDB.GetState(addr, coinID).Big()
-}
-
-// GetState retrieves a value from the given account's storage trie.
-func (s *StateDB) GetState(addr common.Address, hash common.Hash) common.Hash {
-	NormalizeStateKey(&hash)
-	return s.StateDB.GetState(addr, hash)
-}
-
-// AddBalance adds amount to the account associated with addr.
-func (s *StateDB) AddBalanceMultiCoin(addr common.Address, coinID common.Hash, amount *big.Int) {
-	if amount.Sign() == 0 {
-		s.AddBalance(addr, new(uint256.Int)) // used to cause touch
-		return
-	}
-	if !ethstate.GetExtra(s.StateDB, customtypes.IsMultiCoinPayloads, addr) {
-		ethstate.SetExtra(s.StateDB, customtypes.IsMultiCoinPayloads, addr, true)
-	}
-	newAmount := new(big.Int).Add(s.GetBalanceMultiCoin(addr, coinID), amount)
-	NormalizeCoinID(&coinID)
-	s.StateDB.SetState(addr, coinID, common.BigToHash(newAmount))
-}
-
-// SubBalance subtracts amount from the account associated with addr.
-func (s *StateDB) SubBalanceMultiCoin(addr common.Address, coinID common.Hash, amount *big.Int) {
-	if amount.Sign() == 0 {
-		return
-	}
-	// Note: It's not needed to set the IsMultiCoin (extras) flag here, as this
-	// call would always be preceded by a call to AddBalanceMultiCoin, which would
-	// set the extra flag. Seems we should remove the redundant code.
-	if !ethstate.GetExtra(s.StateDB, customtypes.IsMultiCoinPayloads, addr) {
-		ethstate.SetExtra(s.StateDB, customtypes.IsMultiCoinPayloads, addr, true)
-	}
-	newAmount := new(big.Int).Sub(s.GetBalanceMultiCoin(addr, coinID), amount)
-	NormalizeCoinID(&coinID)
-	s.StateDB.SetState(addr, coinID, common.BigToHash(newAmount))
-}
-
-func (s *StateDB) SetState(addr common.Address, key, value common.Hash) {
-	NormalizeStateKey(&key)
-	s.StateDB.SetState(addr, key, value)
-}
-
-// SetTxContext sets the current transaction hash and index which are
-// used when the EVM emits new state logs. It should be invoked before
-// transaction execution.
-func (s *StateDB) SetTxContext(thash common.Hash, ti int) {
-	s.thash = thash
-	s.txIndex = ti
-	s.StateDB.SetTxContext(thash, ti)
-}
```

Comment:
This function was the only reason this struct needed to exist.

Specifically, this function could not be moved to the `extstate.StateDB` implementation because the statedb is wrapped _after_ `SetTxContext` is called.

This function supported the `GetTxHash()` function below.

LibEVM added a `TxHash()` function (to align with the `TxIndex()` function) which allows for us to use the `TxHash()` function rather than the `GetTxHash()` function, which meant that we could remove this entirely.